src/compiler/checker.ts

```diff
@@ -15050,6 +15050,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (isLiteralType(type) || isPatternLiteralType(type)) {
                 checkFlags |= CheckFlags.HasLiteralType;
             }
+            if (isUnion && (type.flags & TypeFlags.Primitive)) {

```

Comment:
Why do we need to check `isUnion`?

Reply:
plot twist: I don't need it

Reply:
I added it hoping to improve the situation with those weird failing baselines, then forgot to remove it